### PR TITLE
Fix/component mesh colocation filter with unique vertex

### PR DESCRIPTION
**Pull request by MelchiorSchuh**

### CODE REVIEW

File: src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp [hunk 1]
```diff
@@ -43,6 +43,67 @@
...
+            const geode::ComponentID& component_id,
+            const std::vector< std::vector< geode::index_t > >&
+                colocated_points_groups )
+    {
+        std::vector< std::vector< geode::index_t > >
+            new_colocated_points_groups;
+        for( const auto& point_group : colocated_points_groups )
+        {
+            auto filtered_colocation =
+                absl::FixedArray< bool >( point_group.size() - 1, true );
```
Comment (BotellaA) [inline src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp:58]:
```suggestion
            absl::FixedArray< bool > filtered_colocation
                ( point_group.size() - 1, true );
```
Why not like that?

File: src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp [hunk 2]
```diff
@@ -43,6 +43,67 @@
...
+            {
+                if( filtered_colocation[pt_id] )
+                {
+                    colocated_points.push_back( point_group[pt_id + 1] );
+                }
+            }
+            if( !colocated_points.empty() )
+            {
+                colocated_points.push_back( point_group[0] );
+                new_colocated_points_groups.push_back(
```
Comment (BotellaA) [inline src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp:89]:
```suggestion
                new_colocated_points_groups.emplace_back(
```

File: src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp [hunk 3]
```diff
@@ -43,6 +43,67 @@
...
+                    std::move( colocated_points ) );
+            }
+        }
+        return new_colocated_points_groups;
+    }
+
+    geode::index_t nb_points(
+        const std::vector< std::vector< geode::index_t > >& points_groups )
```
Comment (BotellaA) [inline src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp:97]:
```suggestion
        absl::Span< const std::vector< geode::index_t > > points_groups )
```

File: src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp [hunk 4]
```diff
@@ -96,8 +163,12 @@ namespace
                 model, verbose );
         for( const auto& block : model.blocks() )
         {
-            const geode::SolidMeshColocation3D inspector{ block.mesh(), false };
-            if( inspector.mesh_has_colocated_points() )
+            const geode::SolidMeshColocation3D inspector{ block.mesh(),
+                verbose };
```
Comment (BotellaA) [inline src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp:167]: Why verbose and not false?

File: src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp [hunk 5]
```diff
@@ -172,8 +249,12 @@ namespace
                 model, verbose );
         for( const auto& block : model.blocks() )
         {
-            const geode::SolidMeshColocation3D inspector{ block.mesh(), false };
-            const auto nb_colocated = inspector.nb_colocated_points();
+            const geode::SolidMeshColocation3D inspector{ block.mesh(),
+                verbose };
```
Comment (BotellaA) [inline src/geode/inspector/criterion/colocation/component_meshes_colocation.cpp:253]: here too?

**Review by BotellaA — APPROVED** [review]

**Review by panquez — APPROVED** [review]

**Pull request merged**